src/App.js

```diff
@@ -56,29 +93,42 @@ function App() {
     }
   }, [currZip, currRange]);
 
+  // frm: Filters the events when there are new events from the API or when user changes filtering criteria
+  useEffect(() => {
+    let newFilteredEvents = events;
+    if (events !== null) {
```

Comment:
Interesting.  I have not heard that particular preference before - what motivates it?  Always happy to learn a new pattern...

What motivated my pattern was the desire to avoid multiple return-stmts (which like break-stmts can lead to hard-to-understand code in some cases).  But again - I am happy to learn new ways ;-)

Reply:
Oh interesting. I prefer early return and immutability over all.

In this case, the top thing feels like a guard to me (do x if you got nothing), so I like to get that out of the way, and then continue onto the meaningful business logic without having to think about how variables might be getting reassigned.

https://refactoring.guru/replace-nested-conditional-with-guard-clauses
https://github.com/airbnb/javascript#references

Reply:
Warning - opinions follow (at length):

In reference to your first link (replace-nested-conditional-with-guard-clauses), I did not buy his aesthetic argument that a sequence of nested if-statements is the problem, because from a visual POV, a sequence of if () else if()  else if() that are arranged one on top of each other is just as clear visally as a sequence of if () return; if () return; etc.   

I am more sympathetic to his comment (and yours): "Try to rid the code of side effects".  This is a very good idea in general, but to be honest, I find it kind of laughable in the Javascript, HTML, CSS world since global state is pervasive in that world.

I got the "only have a single return statement" principle from an old book called The Science of Programming (https://www.amazon.com/Science-Programming-Monographs-Computer/dp/0387964800) which was written by someone interested in proving programs "correct".  His point was that programmers could write programs that were easier to prove by a computer and those programs would also be easier to understand by humans.  What I took away from that book was that a "good" program/function/block was one that proceeded in sequence establishing facts with a single return-stmt at the end.  This allows a computer/human to follow the logic of the program.

However, in this instance, having all of the return-stmts so close together guarded with simple conditions (facts) argues against being dogmatic about a single return.

As for your second link - I totally agree - thanks for sharing it.

Reply:
I replaced my code with yours;  ;-)